File: Python/Intersection of Two Linked Lists.py
class Solution:
    # @param two ListNodes
    # @return the intersected ListNode
    def getIntersectionNode(self, headA, headB):
        nodeA = headA
        nodeB = headB

        while nodeA != nodeB:
            try:
                nodeA = nodeA.next
            except:
                nodeA = headB

            try:
                nodeB = nodeB.next
            except:
                nodeB = headA
        return nodeA

File: Java/src/Intersection of Two Linked Lists .java
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        ListNode a = headA;
        ListNode b = headB;

        while (a != b){
            if (a!= null){
                a = a.next;
            }
            else{
                a = headB;
            }
           if (b!= null){
                b = b.next;
            }
            else{
                b = headA;
            }
        }
        return a;

    }
}